pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -2,138 +2,252 @@
 // under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
-//go:build !windows
 
-//nolint:revive // TODO(PLINT) Fix revive linter
+//nolint:revive // TODO(AGENTRUN) Fix revive linter
 package disk
 
 import (
-	"regexp"
+	"errors"
+	"fmt"
+	"path/filepath"
 	"strings"
 
-	yaml "gopkg.in/yaml.v2"
-
 	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
 	"github.com/DataDog/datadog-agent/pkg/collector/check"
-	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/metrics/servicecheck"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/option"
+	"github.com/mitchellh/mapstructure"
+	yaml "gopkg.in/yaml.v2"
+
+	"regexp"
+
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	gopsutil_disk "github.com/shirou/gopsutil/v4/disk"
 )
 
 const (
-	// CheckName is the name of the check
 	CheckName   = "disk"
 	diskMetric  = "system.disk.%s"
 	inodeMetric = "system.fs.inodes.%s"
 )
 
-type diskConfig struct {
-	useMount             bool
-	excludedFilesystems  []string
-	excludedDisks        []string
-	excludedDiskRe       *regexp.Regexp
-	tagByFilesystem      bool
-	excludedMountpointRe *regexp.Regexp
-	allPartitions        bool
-	deviceTagRe          map[*regexp.Regexp][]string
-}
-
-func (c *Check) excludeDisk(mountpoint, device, fstype string) bool {
-
-	// Hack for NFS secure mounts
-	// Secure mounts might look like this: '/mypath (deleted)', we should
-	// ignore all the bits not part of the mountpoint name. Take also into
-	// account a space might be in the mountpoint.
-	mountpoint = strings.Split(mountpoint, " ")[0]
+var (
+	DiskPartitions = gopsutil_disk.Partitions
+	DiskUsage      = gopsutil_disk.Usage
+	DiskIOCounters = gopsutil_disk.IOCounters
+)
```

Comment:
Could we add make these vars private? You need to change the package name in  `pkg/collector/corechecks/system/disk/disk/disk_test.go` from `disk_test` to `disk`. But making them private is a good practice if we do not intent on using them outside of the disk check 😄